### PR TITLE
MOB-4305 Navigation bar added to the Checkout scene. This required to update N…

### PR DESCRIPTION
**Pull request by AleksanderWedrychowskiKarhoo**
…avigationController implementation, so it is able to adjust its design on its-own.

## JIRA
[MOB-4305](https://karhoo.atlassian.net/browse/MOB-4305)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: KarhooUISDK/Controllers/NavigationController.swift [hunk 1]
```diff
@@ -11,10 +11,51 @@ import UIKit
...
+        case primary
+        /// Style uses background color as bar tint color
+        case secondary
+
+        var backgroundColor: UIColor {
+            switch self {
+            case .primary:
+                return KarhooUI.colors.primary
+            case .secondary:
+                return KarhooUI.colors.white
```
Comment (dianakarh) [inline KarhooUISDK/Controllers/NavigationController.swift:25]: We'll need to rename white and black as the names are confusing when in dark mode. Maybe `defaultBackground`? Which would be white in light mode and black in dark mode? Not for now, but it's a discussion we need to have.

**Review by dianakarh — CHANGES_REQUESTED** [review]
Still needs:
- translations
- a fix in the ride details screen

File: KarhooUISDK/Controllers/NavigationController.swift (hunk 1 above)
Comment (AleksanderWedrychowskiKarhoo) [reply KarhooUISDK/Controllers/NavigationController.swift:25]: I think the idea is so we have `white` that is acctualy always white, no matter on light/dark mode. In this case the background color needs to be changed to `background1` or any else, but we need to update the design to do it properly.

**Comment by AleksanderWedrychowskiKarhoo** [comment]
Simulator Screen Shot - iPhone 13 mini - 2022-08-23 at 10 42 15

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 2]
```diff
@@ -70,6 +70,7 @@ public enum UITexts {
         public static let sortBy = "Text.Generic.SortBy".localized
         public static let driverArrival = "Text.Generic.DriverArrival".localized
         public static let all = "Text.Generic.all".localized
+        public static let checkout = "Text.Generic.Checkout".localized
```
[the remark this answers is not in the document]
Comment (AleksanderWedrychowskiKarhoo) [reply KarhooUISDK/CommonUI/UITexts.swift:73]: Updated :)

File: KarhooUISDK/Screens/Rides/RideDetails/RideDetailsViewController.swift [hunk 3]
```diff
@@ -174,7 +174,7 @@ final class RideDetailsViewController: UIViewController, RideDetailsView {
         private
         func embedInNavigationController(_ vc: UIViewController,
                                          closeCallback: @escaping ScreenResultCallback<RideDetailsAction>) -> Screen {
-            let navigationController = NavigationController(rootViewController: vc)
+            let navigationController = NavigationController(rootViewController: vc, style: .secondary)
```
[the remark this answers is not in the document]
Comment (AleksanderWedrychowskiKarhoo) [reply KarhooUISDK/Screens/Rides/RideDetails/RideDetailsViewController.swift:177]: Several small adjustments added:
- back arrow tint changed to align with title color
- back item title removed (in this case "Rides")
- Ride details view shadow aligned with new design (opacity 0.2, radius 3)
- Ride details view corner radius aligned with new design (8)

File: KarhooUISDK/Controllers/NavigationController.swift [hunk 4]
```diff
@@ -23,6 +64,34 @@ final class NavigationController: UINavigationController {
     private func setupDelegate() {
         delegate = self
     }
+
+    private func setupDesign() {
+        if #available(iOS 13.0, *) {
```
Comment (dianakarh) [inline KarhooUISDK/Controllers/NavigationController.swift:69]: maybe merge the bump to ios 13 into this branch to cleanup the availability check?

File: KarhooUISDK/Extensions/UIViewController+Extensions.swift [hunk 5]
```diff
@@ -12,10 +12,6 @@ import UIKit
 public extension UIViewController {
     func forceLightMode() {
         if #available(iOS 13.0, *) {
```
Comment (dianakarh) [inline KarhooUISDK/Extensions/UIViewController+Extensions.swift:14]: another availability check here

File: KarhooUISDK/Translations/it.lproj/Localizable.strings [hunk 6]
```diff
@@ -618,9 +620,9 @@
 
 "Text.QuoteCategory.Electric" = "Elettrico";
 
-"Text.QuoteCategory.Exec" = "Esecutivo";
+"Text.QuoteCategory.Exec" = "Executive";
 
-"Text.QuoteCategory.Executive" = "Esecutivo";
+"Text.QuoteCategory.Executive" = "Executive";
```
Comment (dianakarh) [inline KarhooUISDK/Translations/it.lproj/Localizable.strings:625]: looks like regression. @olivierkarhoo do you know anything about this change?

File: KarhooUISDK/Translations/it.lproj/Localizable.strings [hunk 7]
```diff
@@ -897,7 +899,7 @@
 /* Vehicle tags */
 "Text.VehicleTag.Electric" = "Elettrico";
 
-"Text.VehicleTag.Executive" = "Esecutivo";
+"Text.VehicleTag.Executive" = "Executive";
```
Comment (dianakarh) [inline KarhooUISDK/Translations/it.lproj/Localizable.strings:902]: here too @olivierkarhoo

File: KarhooUISDK/Translations/it.lproj/Localizable.strings (hunk 6 above)
Comment (olivierkarhoo) [reply KarhooUISDK/Translations/it.lproj/Localizable.strings:625]: Yes executive is correct, I changed it after asking to Italian native person yesterday

Comment (olivierkarhoo) [reply KarhooUISDK/Translations/it.lproj/Localizable.strings:625]: @dianakarh

**Review by dianakarh — APPROVED** [review]

**Review by dianakarh — APPROVED** [review]

**Pull request merged**